server/src/slicec_ext/diagnostic_ext.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) ZeroC, Inc.
+
+use slicec::diagnostics::{Diagnostic, DiagnosticLevel, Note};
```

Comment:
A general comment about this structure:
I think using the extension function pattern is overkill here.

You could remove the `slicec_ext` directory and module structure, and delete these 2 `Ext` traits, and instead just use normal functions:

`try_into_lsp_diagnostic(diagnostic: &Diagnostic) -> Option<...>`
`try_into_lsp_info...(note: &Note) -> ...`

Having standalone conversion functions is totally fine in Rust,
and should save you alot of code compared to using extension functions.